sys/src/9/amd64/multiboot.c

```diff
@@ -103,6 +105,29 @@ mbtypename(int type)
 	return "(unknown)";
 }
 
+static i32
+mbmoduleread(Chan *c, void *buf, i32 len, i64 off)
+{
+	if (!initrd) {
+		error("mbmoduleread: module not found");
+	}
+
+	u64 modlen = initrd->modend - initrd->modstart;
+	void *data = (void*)(u64)initrd->modstart;
+
+	if (off < 0 || off > modlen) {
```

Comment:
see readmem, I think it does what you want.

Reply:
yup - thanks